src/views/Deposit/components/EnterAmount/index.tsx

```diff
@@ -17,7 +22,89 @@ import {
 } from './components';
 import { H2, P2 } from 'components/Typography';
 
-export const EnterAmount: React.FC = () => {
+// Constants
+import { TOKEN_GEYSER_ADDRESS, WPOKT_ADDRESS } from 'constants/index';
+
+// Contexts
+import { BalanceContext } from 'contexts/Balance';
+import { Web3Context } from 'contexts/Web3';
+
+// Utils
+import { approve, bnToDec, decToBn, getAllowance, getNotification, stake } from 'utils';
+
+interface IEnterAmount {
+  farmSelected: boolean;
+  setFarmSelected: React.Dispatch<React.SetStateAction<boolean>>;
+}
+
+export const EnterAmount: React.FC<IEnterAmount> = ({ farmSelected, setFarmSelected }) => {
+  const { wpoktBalance } = React.useContext(BalanceContext);
+  const [isApproved, setIsApproved] = React.useState<boolean>(true);
+  const [isDisabled, setIsDisabled] = React.useState<boolean>(true);
+  const [wpoktInputValue, setWpoktInputValue] = React.useState<string>('');
+  const { address, onboard, provider, notify, signer } = React.useContext(Web3Context);
+
+  // Should probably a separated hook
+  const useApproval = async () => {
+    if (address && provider && !isDisabled && farmSelected) {
+      const allowance = await getAllowance(address, TOKEN_GEYSER_ADDRESS, WPOKT_ADDRESS, provider);
+      if (+decToBn(+wpoktInputValue) > +allowance) {
+        setIsApproved(true);
+      } else {
+        setIsApproved(false);
+      }
+    }
+  };
+  useApproval();
+
+  // Effects
+  React.useEffect(() => {
+    if (wpoktInputValue === '' || wpoktInputValue === '0' || !farmSelected) {
+      setIsDisabled(true);
+    } else {
+      setIsDisabled(false);
+    }
+  }, [wpoktInputValue, farmSelected]);
+
+  // Handlers
```

Comment:
:)

Reply:
Fixed in 4bf8642d6e8b1d3eab7e08c97ee4512e7d289920